### PR TITLE
Remove dummy net code

### PR DESCRIPTION
**Pull request by youkaichao**
- [ ] I have marked all applicable categories:
    + [ ] exception-raising fix
    + [ ] algorithm implementation fix
    + [ ] documentation modification
    + [ ] new feature
- [ ] If applicable, I have mentioned the relevant/related issue(s)

Less important but also useful:

- [ ] I have visited the [source website], and in particular read the [known issues]
- [ ] I have searched through the [issue tracker] and [issue categories] for duplicates
- [ ] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, torch, sys
  print(tianshou.__version__, torch.__version__, sys.version, sys.platform)
  ```

  [source website]: https://github.com/thu-ml/tianshou
  [known issues]: https://github.com/thu-ml/tianshou/#faq-and-known-issues
  [issue categories]: https://github.com/thu-ml/tianshou/projects/2
  [issue tracker]: https://github.com/thu-ml/tianshou/issues?q=

### CODE REVIEW

**Comment by duburcqa** [comment]
Why not to keep the original names `Actor` / `Critic`? I think it is clearer. I don't see any advantage of introducing extra postfix `Head`. It is not confusing since `Net` is clearly not an Actor or Critic on its own.

**Comment by youkaichao** [comment]
> Why not to keep the original names `Actor` / `Critic`? I think it is clearer. I don't see any advantage of introducing extra postfix `Head`. It is not confusing since `Net` is clearly not an Actor or Critic on its own.

Good idea, solved in new commit.

**Review by duburcqa — APPROVED** [review]

File: examples/ant_v2_ddpg.py [hunk 1]
```diff
@@ -57,14 +57,12 @@ def test_ddpg(args=get_args()):
...
-        args.max_action, args.device
-    ).to(args.device)
+    net = Net(args.layer_num, args.state_shape, device=args.device)
+    actor = Actor(net, args.action_shape, args.max_action,
+                  args.device).to(args.device)
     actor_optim = torch.optim.Adam(actor.parameters(), lr=args.actor_lr)
-    critic = Critic(
-        args.layer_num, args.state_shape, args.action_shape, args.device
-    ).to(args.device)
+    net = Net(args.layer_num, args.state_shape, args.action_shape, concat=True)
```
Comment (Trinkle23897) [inline examples/ant_v2_ddpg.py]: missing device

File: examples/ant_v2_sac.py [hunk 2]
```diff
@@ -58,17 +58,19 @@ def test_sac(args=get_args()):
...
     actor = ActorProb(
-        args.layer_num, args.state_shape, args.action_shape,
+        net, args.action_shape,
         args.max_action, args.device, unbounded=True
     ).to(args.device)
     actor_optim = torch.optim.Adam(actor.parameters(), lr=args.actor_lr)
+    net = Net(args.layer_num, args.state_shape, args.action_shape, concat=True)
     critic1 = Critic(
-        args.layer_num, args.state_shape, args.action_shape, args.device
+        net, args.device
```
Comment (Trinkle23897) [inline examples/ant_v2_sac.py]: You can merge Critic(...) to one line. Same as below.

File: examples/ant_v2_td3.py [hunk 3]
```diff
@@ -60,17 +60,19 @@ def test_td3(args=get_args()):
...
     test_envs.seed(args.seed)
     # model
+    net = Net(args.layer_num, args.state_shape, device=args.device)
     actor = Actor(
-        args.layer_num, args.state_shape, args.action_shape,
+        net, args.action_shape,
         args.max_action, args.device
     ).to(args.device)
     actor_optim = torch.optim.Adam(actor.parameters(), lr=args.actor_lr)
+    net = Net(args.layer_num, args.state_shape, args.action_shape, concat=True)
```
Comment (Trinkle23897) [inline examples/ant_v2_td3.py]: device

File: examples/halfcheetahBullet_v0_sac.py [hunk 4]
```diff
@@ -66,17 +66,19 @@ def test_sac(args=get_args()):
...
     test_envs.seed(args.seed)
     # model
+    net = Net(args.layer_num, args.state_shape, device=args.device)
     actor = ActorProb(
-        args.layer_num, args.state_shape, args.action_shape,
+        net, args.action_shape,
         args.max_action, args.device, unbounded=True
     ).to(args.device)
     actor_optim = torch.optim.Adam(actor.parameters(), lr=args.actor_lr)
+    net = Net(args.layer_num, args.state_shape, args.action_shape, concat=True)
```
Comment (Trinkle23897) [inline examples/halfcheetahBullet_v0_sac.py]: device

**Review by Trinkle23897 — APPROVED** [review]

File: examples/ant_v2_td3.py [hunk 5]
```diff
@@ -60,18 +60,17 @@ def test_td3(args=get_args()):
...
-        args.layer_num, args.state_shape, args.action_shape, args.device
-    ).to(args.device)
+    net = Net(args.layer_num, args.state_shape,
+              args.action_shape, concat=True, device=args.device)
+    critic1 = Critic(net, args.device).to(args.device)
     critic1_optim = torch.optim.Adam(critic1.parameters(), lr=args.critic_lr)
-    critic2 = Critic(
-        args.layer_num, args.state_shape, args.action_shape, args.device
-    ).to(args.device)
+    critic2 = Critic(net, args.device).to(args.device)
```
Comment (Trinkle23897) [inline examples/ant_v2_td3.py:73]: Actually, TD3 cannot support share-critic, which causes #209.
